src/helpers.ts

```diff
@@ -329,3 +329,43 @@ export function entries(obj: any): [string, any][] {
 		resArray[i] = [ownProps[i], obj[ownProps[i]]];
 	return resArray;
 }
+
+function callSplice(arr: any[], index: number, removeCount: number, addItems: any[]): void {
+	if (removeCount || addItems.length > 0) {
+		var addItemsArgs = [index, removeCount].concat(addItems);
+		arr.splice.apply(arr, addItemsArgs);
+	}
+}
+
+export function updateArray(array: any[], values: any[]): void {
```

Comment:
Removed the unused event tracking code from `updateArray` and moved it from the "observable-array" file to "helpers" since it no longer contains any "observable" specific code.